src/chart/generateCategoricalChart.tsx

```diff
@@ -1934,14 +1955,15 @@ export const generateCategoricalChart = ({
         return null;
       }
 
-      const { isTooltipActive, activeCoordinate, activePayload, activeLabel, offset } = this.state;
+      const { isTooltipActive, activeCoordinate, activePayload, activeLabel, offset, activeDataKey } = this.state;
 
       return cloneElement(tooltipItem, {
         viewBox: { ...offset, x: offset.left, y: offset.top },
         active: isTooltipActive,
         label: activeLabel,
         payload: isTooltipActive ? activePayload : [],
         coordinate: activeCoordinate,
+        tooltipDatakey: activeDataKey,
```

Comment:
casing inconsistency `tooltipDatakey` vs `activeDataKey`
```suggestion
        tooltipDataKey: activeDataKey,
```

Reply:
that is my preference since it is two separate words words